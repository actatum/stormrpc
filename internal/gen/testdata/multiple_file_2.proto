syntax = "proto3";

package test;

option go_package = "./test;prototest";

service Pet {
    rpc SayPet (PetRequest) returns (PetReply) {}
}

message PetRequest {
    string name = 1;
}

message PetReply {
    string message = 1;
}
