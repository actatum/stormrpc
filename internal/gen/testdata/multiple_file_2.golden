// Code generated by protoc-gen-stormrpc. DO NOT EDIT.

package prototest

import (
	context "context"
	fmt "fmt"
	stormrpc "github.com/actatum/stormrpc"
)

// PetClient is the client API for Pet service.
type PetClient interface {
	SayPet(ctx context.Context, in *PetRequest, opts ...stormrpc.CallOption) (*PetReply, error)
}

type petClient struct {
	c *stormrpc.Client
}

func NewPetClient(c *stormrpc.Client) PetClient {
	return &petClient{c}
}

func (c *petClient) SayPet(ctx context.Context, in *PetRequest, opts ...stormrpc.CallOption) (*PetReply, error) {
	var out PetReply
	r, err := stormrpc.NewRequest("rpc.Pet.SayPet", in, stormrpc.WithEncodeProto())
	if err != nil {
		return nil, err
	}

	resp := c.c.Do(ctx, r, opts...)
	if resp.Err != nil {
		return nil, resp.Err
	}

	if err = resp.Decode(&out); err != nil {
		return nil, err
	}

	return &out, nil
}

// PetServer is the server API for Pet service.
type PetServer interface {
	SayPet(context.Context, *PetRequest) (*PetReply, error)
}

func RegisterPetServer(s *stormrpc.Server, srv PetServer) {
	for _, handler := range petHandlers {
		handler.SetService(srv)
		s.Handle(handler.Route(), handler.HandlerFunc())
	}
}

type _Pet_SayPet_Handler struct {
	route string
	svc   interface{}
}

func (h *_Pet_SayPet_Handler) HandlerFunc() stormrpc.HandlerFunc {
	return func(ctx context.Context, r stormrpc.Request) stormrpc.Response {
		var in PetRequest
		if err := r.Decode(&in); err != nil {
			return stormrpc.NewErrorResponse(r.Reply, fmt.Errorf("error decoding request"))
		}

		out, err := h.svc.(PetServer).SayPet(ctx, &in)
		if err != nil {
			return stormrpc.NewErrorResponse(r.Reply, err)
		}

		resp, err := stormrpc.NewResponse(r.Reply, out, stormrpc.WithEncodeProto())
		if err != nil {
			return stormrpc.NewErrorResponse(r.Reply, err)
		}

		return resp
	}
}

func (h *_Pet_SayPet_Handler) Route() string {
	return h.route
}

func (h *_Pet_SayPet_Handler) SetService(svc interface{}) {
	h.svc = svc
}

var petHandlers = []handler{
	&_Pet_SayPet_Handler{route: "rpc.Pet.SayPet"},
}
